backend/src/bundles/user-details/user-details.controller.ts

```
@@ -358,6 +371,46 @@ class UserDetailsController extends ControllerBase {
             payload: await this.userDetailsService.update(options.body),
         };
     }
+
+    /**
+     * @swagger
+     * /user-details/{userId}:
+     *    get:
+     *      tags: [User Details]
+     *      description: Returns user details by user ID
+     *      security:
+     *        - bearerAuth: []
+     *      parameters:
+     *        - in: path
+     *          name: userId
+     *          required: true
+     *          description: User ID to fetch details for
+     *          schema:
+     *            type: string
+     *            format: uuid # Example: '550e8400-e29b-41d4-a716-446655440000'
+     *      responses:
+     *        200:
+     *          description: Successful operation
+     *          content:
+     *            application/json:
+     *              schema:
+     *                $ref: '#/components/schemas/UserDetails'
+     */
+
+    private async findByUserId(
+        options: ApiHandlerOptions<{
+            params: Partial<UserDetailsFindRequestDto>;
+        }>,
+    ): Promise<ApiHandlerResponse> {
+        const { userId } = options.params;
+
+        return {
+            status: HttpCode.OK,
+            payload: userId
+                ? await this.userDetailsService.findByUserId(userId)
+                : null,
```

If user didn't find please throw 404 Not Found Error